qa/workunits/rbd/rbd_mirror_helpers.sh

```diff
@@ -795,6 +875,53 @@ get_newest_mirror_snapshot()
         ${log} || true
 }
 
+get_newest_mirror_snapshot_id_on_primary()
+{
+    local primary_cluster=$1
+    local image_spec=$2
+    local -n _snap_id=$3
+
+    run_cmd "rbd --cluster ${primary_cluster} snap list --all ${image_spec} --format xml --pretty-format"
+    _snap_id=$(xmlstarlet sel -t -v "(//snapshots/snapshot[namespace/complete='true']/id)[last()]" "$CMD_STDOUT" )
+}
+
+test_snap_present()
+{
+    local secondary_cluster=$1
+    local image_spec=$2
+    local snap_id=$3
+    local expected_snap_count=$4
+
+    run_cmd "rbd --cluster ${secondary_cluster} snap list -a ${image_spec} --format xml --pretty-format"
+    test "${expected_snap_count}" = "$(xmlstarlet sel -t -v "count(//snapshots/snapshot/namespace[primary_snap_id='${snap_id}'])" < "$CMD_STDOUT")" || { fail; return 1; }
+}
+
+wait_for_test_snap_present()
+{
+    local secondary_cluster=$1
+    local image_spec=$2
+    local snap_id=$3
+    local test_snap_count=$4
+    local s
+
+    for s in 0.1 1 2 4 8 8 8 8 8 8 8 8 16 16 32 32; do
+        sleep ${s}
+        test_snap_present "${secondary_cluster}" "${image_spec}" "${snap_id}" "${test_snap_count}" && return 0
+    done
+
+    fail "wait for count of snaps with id ${snap_id} to be ${test_snap_count} failed on ${secondary_cluster}"
+    return 1
+}
+
+wait_for_snap_id_present()
```

Comment:
Instead of these three helpers, how about a helper like,
```
wait_for_primary_snap_id_present_in_secondary()
{
    local secondary_cluster=$1
    local image_spec=$2
    local primary_snap_id=$3
    local s

    for s in 0.1 1 2 4 8 8 8 8 8 8 8 8 16 16 32 32; do
        rbd --cluster ${secondary_cluster} snap list -a ${image_spec} --format xml | \
            xmlstarlet sel -t -c "//snapshots/snapshot/namespace[primary_snap_id='${snap_id}']" && return 0
    done

    return 1
}
```
@VinayBhaskar-V @idryomov what do you think?

Reply:
And now I see this helper called `get_primary_snap_id_for_newest_mirror_snapshot_on_secondary`  in `wip-rbd-cgsm-base` branch that gets "the primary_snap_id for the most recent complete snap on the secondary cluster".  IMO, the helper should have included "complete" in its name.  It looks like some of the helpers in `wip-rbd-cgsm-base` branch need to be cleaned up and cannot be just copied over.